From c601984805311cdb58e7321d6f69b6280e8718a4 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:53 -0400
Subject: [net] usb/cdc_ncm: remove redundant "intf" field

Message-id: <1410887463-20674-162-git-send-email-dzickus@redhat.com>
Patchwork-id: 95130
O-Subject: [RHEL7 PATCH 161/591] net: cdc_ncm: remove redundant "intf" field
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 3e515665a76ad8f60a1c05968cc6a5b2f2701171
Author: Bjorn Mork <bjorn@mork.no>
Date:   Fri Nov 1 11:16:40 2013 +0100

    net: cdc_ncm: remove redundant "intf" field

    This is always a duplicate of the "control" field. It causes
    confusion wrt intf_data updates and cleanups.

    Cc: Alexey Orishko <alexey.orishko@gmail.com>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/usb/cdc_ncm.c b/drivers/net/usb/cdc_ncm.c
index 43afde8..7c5c762 100644
--- a/drivers/net/usb/cdc_ncm.c
+++ b/drivers/net/usb/cdc_ncm.c
@@ -382,7 +382,6 @@ int cdc_ncm_bind_common(struct usbnet *dev, struct usb_interface *intf, u8 data_
  len = intf->cur_altsetting->extralen;
 
  ctx->udev = dev->udev;
- ctx->intf = intf;
 
  /* parse through descriptors associated with control interface */
  while ((len > 0) && (buf[0] > 2) && (buf[0] <= len)) {
@@ -489,7 +488,6 @@ advance:
 
  usb_set_intfdata(ctx->data, dev);
  usb_set_intfdata(ctx->control, dev);
- usb_set_intfdata(ctx->intf, dev);
 
  if (ctx->ether_desc) {
   temp = usbnet_get_ethernet_addr(dev, ctx->ether_desc->iMACAddress);
@@ -553,7 +551,7 @@ void cdc_ncm_unbind(struct usbnet *dev, struct usb_interface *intf)
   ctx->control = NULL;
  }
 
- usb_set_intfdata(ctx->intf, NULL);
+ usb_set_intfdata(intf, NULL);
  cdc_ncm_free(ctx);
 }
 EXPORT_SYMBOL_GPL(cdc_ncm_unbind);
diff --git a/include/linux/usb/cdc_ncm.h b/include/linux/usb/cdc_ncm.h
index 89f0bbc..c14e00f 100644
--- a/include/linux/usb/cdc_ncm.h
+++ b/include/linux/usb/cdc_ncm.h
@@ -103,7 +103,6 @@ struct cdc_ncm_ctx {
  struct usb_host_endpoint *in_ep;
  struct usb_host_endpoint *out_ep;
  struct usb_host_endpoint *status_ep;
- struct usb_interface *intf;
  struct usb_interface *control;
  struct usb_interface *data;
 
-- 
1.7.1